packages/project-access/src/file/file-search.ts

```diff
@@ -0,0 +1,91 @@
+import type { Editor } from 'mem-fs-editor';
+import { basename, dirname, join, sep } from 'path';
+import { default as find } from 'findit2';
+import { fileExists } from './file-access';
+
+/**
+ * Add missing dump properties
+ */
+declare module 'mem-fs-editor' {
+    type FileMap = { [key: string]: { state: 'modified' | 'deleted' } };
+
+    export interface Editor {
+        dump(cwd: string): FileMap;
+    }
+}
+
+/**
+ * Search for 'filename' starting from 'root' in the list of virtual (not committed) files. Returns array of paths that contain the file.
+ *
+ * @param files - - array of paths that contain the filename found on the filesystem
+ * @param filename - filename to search
+ * @param root - root folder to start search
+ * @param stopFolders - list of folder names to exclude (search doesn't traverse into these folders)
+ * @param fs - mem-fs-editor instance
+ * @returns - enhanced array of paths that contain the filename
+ */
+function checkVirtualFiles(files: string[], filename: string, root: string, stopFolders: string[], fs: Editor) {
+    const memFiles = fs.dump(root);
+    const modified = Object.keys(memFiles)
+        .filter((file) => memFiles[file].state === 'modified' && file.endsWith(filename))
+        .map((file) => dirname(join(root, file)));
+    const ignore = Object.keys(memFiles)
+        .filter(
+            (file) =>
+                memFiles[file].state === 'deleted' ||
+                stopFolders.find((folder) => file.includes(`${sep}${folder}${sep}`))
+        )
+        .map((file) => dirname(join(root, file)))
+        .concat(...modified);
+    return files.filter((match) => !ignore.includes(match)).concat(...modified);
+}
+
+/**
+ * Search for 'filename' starting from 'root'. Returns array of paths that contain the file.
+ *
+ * @param filename - filename to search
+ * @param root - root folder to start search
+ * @param stopFolders - list of folder names to exclude (search doesn't traverse into these folders)
+ * @param fs - optional mem-fs-editor instance
+ * @returns - array of paths that contain the filename
+ */
+export function findFiles(filename: string, root: string, stopFolders: string[], fs?: Editor): Promise<string[]> {
```

Comment:
Does `exclude` sound better than `stopFolders`?

Reply:
Changed it to `excludeFolders`, in JS Doc we also use the word exclude so your proposal makes it consistent.